tests/app/questionnaire_state/test_state_item.py

```diff
@@ -0,0 +1,44 @@
+from unittest import TestCase
+
+from mock import MagicMock
+
+from app.questionnaire_state.state_item import StateItem
+
+
+class TestStateItem(TestCase):
+
+    def test_find_schema_item_returns_if_found(self):
+        # Given
+        schema_item = MagicMock()
+
+        search_schema = MagicMock()
+
+        child = MagicMock()
+        child.schema_item = search_schema
+        state_item = StateItem('item_id', schema_item)
+        state_item.children = [child]
+
+        # When
+        result = state_item.find_state_item(search_schema)
+
+        # Then
+        self.assertEqual(result, child)
+
+    def test_find_schema_item_recursive_call_if_not_found(self):
```

Comment:
Test names might be nicer test_find_schema_item_in_question and test_find_schema_item_in_block as I think that's what it's testing right? Again minor (possibly preference)